opentelemetry-sdk/src/opentelemetry/sdk/trace/_sampling_experimental/_parent_based.py

```diff
@@ -0,0 +1,65 @@
+from typing import Optional, Sequence
+
+from opentelemetry.context import Context
+from opentelemetry.trace import Link, SpanKind, TraceState, get_current_span
+from opentelemetry.util.types import Attributes
+
+from ._composable import ComposableSampler, SamplingIntent
+from ._sampler import ConsistentSampler
+from ._trace_state import OtelTraceState
+from ._util import (
+    INVALID_THRESHOLD,
+    MIN_THRESHOLD,
+    is_valid_threshold,
+)
+
+
+class ConsistentParentBasedSampler(ComposableSampler):
+    def __init__(self, root_sampler: ComposableSampler):
+        self._root_sampler = root_sampler
+        self._description = f"ConsistentParentBasedSampler{{root_sampler={root_sampler.get_description()}}}"
+
+    def sampling_intent(
+        self,
+        parent_ctx: Optional[Context],
+        name: str,
+        span_kind: Optional[SpanKind],
+        attributes: Attributes,
+        links: Optional[Sequence[Link]],
+        trace_state: Optional[TraceState] = None,
+    ) -> SamplingIntent:
+        parent_span = get_current_span(parent_ctx)
+        parent_span_ctx = parent_span.get_span_context()
+        is_root = not parent_span_ctx.is_valid
+        if is_root:
```

Comment:
There is a chance of attribute error, so this could be more like og [ParentBased](https://github.com/open-telemetry/opentelemetry-python/blob/cf31a6967aac76663b4cca023b5d760fd6eb5355/opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py#L349-L351):

```suggestion
        parent_span_ctx = get_current_span(
            parent_ctx
        ).get_span_context()
        if parent_span_ctx is not None and parent_span_ctx.is_valid:
```